### ISSUE TITLE
KeyError: 'cursor' when performing a ScopusSearch

### ISSUE THREAD

**Issue by liogiu2** [pybliometrics-dev/pybliometrics#127]
I'm performing the following query using ScopusSearch: "TITLE (interactive AND management) AND  ( LIMIT-TO ( SUBJAREA ,  "COMP" ) )". If I remove from the query the part where I limit the results to the subject area of computer science, the query works properly. 

I tried a hard fix by using a try/except in the lines that cause problems. The result is that the system ignores the ( LIMIT-TO ( SUBJAREA ,  "COMP" ) ) part of the query.

I'm working on a MacOs Catalina 10.15.1 with pybliometric version 2.2.1

Here the traceback of my call

```
KeyError                                  Traceback (most recent call last)
/.../scopus_search.py in 
---> 17     s = ScopusSearch(final_search_key, verbose=True, refresh=True)

~/anaconda3/envs/ru-env/lib/python3.7/site-packages/pybliometrics/scopus/scopus_search.py in __init__(self, query, refresh, subscriber, view, download, integrity_fields, integrity_action, verbose, **kwds)
    195         Search.__init__(self, query=query, api='ScopusSearch', refresh=refresh,
    196                         count=count, cursor=subscriber, view=view,
--> 197                         download=download, verbose=verbose, **kwds)
    198         self.integrity = integrity_fields or []
    199         self.action = integrity_action

~/anaconda3/envs/ru-env/lib/python3.7/site-packages/pybliometrics/scopus/classes/search.py in __init__(self, query, api, refresh, view, count, max_entries, cursor, download, verbose, **kwds)
     86                 raise ScopusQueryError(text)
     87             if download:
---> 88                 self._json = _parse(res, params, n, api, verbose, **kwds)
     89                 # Finally write out the file
     90                 with open(qfile, 'wb') as f:

~/anaconda3/envs/ru-env/lib/python3.7/site-packages/pybliometrics/scopus/classes/search.py in _parse(res, params, n, api, verbose, **kwds)
    118         n -= params["count"]
    119         if cursor:
--> 120             pointer = res['search-results']['cursor'].get('@next')
    121             params.update({'cursor': pointer})
    122         else:

KeyError: 'cursor'
```

**Comment by Michael-E-Rose** [comment]
Hi @liogiu2, thanks for this elaborate issue.

tl;dr: Refresh the query until it works or assure you have a smaller result set.

Explanation:
The cursor, whose absence raises the KeyError, is a relatively new feature of Scopus to allow fast switching through result sets.  Results sets are usually limited to 250, so more results of a query require a cursor.

It quickly becomes apparent, that the cursor should _always_ be there there, and pybliometrics needs it to cycle through the results sets.  Alas, for problems related to Scopus' servers, it is not.  They are aware of the issue and in June promised to work on this.

As the cursor misses completely idiosyncratically, refreshing helps.  But it's not guaranteed to work, as the probability that the cursor is absent is proportional to the total number of results of a query.  Another option is thus to make queries with less total results.

**Comment by Michael-E-Rose** [comment]
While my answer was valid until very recently, there seems to be something more systematic about the absence of the cursor.  Meaning, it's not there anymore.

I am investigating the issue; in particular whether it's gone for good.

In the meantime, running a query with `cursor=False` is the only thing that works.  It will come at a cost, though: Queries with more than 5000 results are invalid, and it takes longer to download the data.

**Comment by Michael-E-Rose** [comment]
It's a bug.  Scopus' US team is working on it.
